eclipse/che
# Provides direct access to the kubernetes API

### What does this PR do?

The endpoint is available at `/api/unsupported/k8s` and should be restricted only to OpenShift with OpenShift OAuth.


### What issues does this PR fix or reference?
#18326 

### How to test this PR?
1. Run Che on OpenShift with OpenShift OAuth on
1. Log in as a non-admin user
1. Start a workspace
1. Obtain an JWT access token for the logged in user (best done by looking at the traffic from the browser to the backend)
1. Trying
`curl -k -v -H "Authorization: Bearer  <<ACCESS_TOKEN>>" https://<<CHE_HOST>>/api/unsupported/k8s/api/v1/namespaces/<<WORKSPACE_NAMESPACE>>/serviceaccounts` should return a list of service accounts in the workspace namespace of given user.
Trying `curl -k -v -H "Authorization: Bearer  <<ACCESS_TOKEN>>" https://<<CHE_HOST>>/api/unsupported/k8s/api/v1/namespaces/default/serviceaccounts` should be unauthorized.


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
